apps/api/tests/test_embedded_widget_auth.py

```diff
@@ -0,0 +1,332 @@
+    """Test that unauthorized domain is rejected by permission check."""
+    api_client.session["auth_data"] = {
+        "experiment_id": str(embedded_session.experiment.public_id)
+    }
+    api_client.session.save()
+
+    url = reverse(
+        "api:chat:send-message", kwargs={"session_id": embedded_session.external_id}
+    )
+    data = {"message": "Hello bot"}
+
+    response = api_client.post(
+        url,
+        data=data,
+        format="json",
+        HTTP_X_EMBED_KEY="test_widget_token_123456789012",
+        HTTP_ORIGIN="https://evil.com",
+    )
+
+    # Should fail permission check
+    assert response.status_code == 403
+
+
+@pytest.mark.django_db()
+def test_send_message_without_origin_header(api_client, embedded_session):
+    """Test that missing Origin/Referer header is rejected by permission check."""
+    api_client.session["auth_data"] = {
+        "experiment_id": str(embedded_session.experiment.public_id)
+    }
+    api_client.session.save()
+
+    url = reverse(
+        "api:chat:send-message", kwargs={"session_id": embedded_session.external_id}
+    )
+    data = {"message": "Hello bot"}
+
+    response = api_client.post(
+        url,
+        data=data,
+        format="json",
+        HTTP_X_EMBED_KEY="test_widget_token_123456789012",
+    )
+
+    # Should fail permission check due to missing origin
+    assert response.status_code == 403
+
+
+@pytest.mark.django_db()
+def test_poll_response_with_valid_embed_key_and_domain(api_client, embedded_session):
+    """Test polling messages with valid X-Embed-Key and allowed domain."""
+    api_client.session["auth_data"] = {
+        "experiment_id": str(embedded_session.experiment.public_id)
+    }
+    api_client.session.save()
+
+    url = reverse(
+        "api:chat:poll-response", kwargs={"session_id": embedded_session.external_id}
+    )
+
+    response = api_client.get(
+        url,
+        HTTP_X_EMBED_KEY="test_widget_token_123456789012",
+        HTTP_ORIGIN="https://example.com",
+    )
+
+    assert response.status_code == 200
+    response_json = response.json()
+    assert "messages" in response_json
+    assert "session_status" in response_json
+
+
+@pytest.mark.django_db()
+def test_legacy_flow_without_embed_key(api_client, experiment):
+    """Test that the legacy flow still works without X-Embed-Key header."""
+    url = reverse("api:chat:start-session")
+    data = {
+        "chatbot_id": experiment.public_id,
+        "session_data": {"source": "widget"},
+    }
+
+    # No X-Embed-Key header - should use legacy flow (SessionAuthentication or unauthenticated)
+    response = api_client.post(url, data=data, format="json")
+
+    assert response.status_code == 201
+    response_json = response.json()
+    assert "session_id" in response_json
+
+
+# Tests for authentication behavior
+
+
+@pytest.mark.django_db()
+def test_authentication_returns_experiment_channel(api_client, embedded_widget_channel):
```

Comment:
This seems to duplicate earlier tests without actually testing anything different